### PR TITLE
[Test] Fix test_dashboard.py in Python 3.10

### LINKED ISSUE #29298: [CI] `linux://python/ray/dashboard:test_dashboard` is failing/flaky on release branch.
- 4b605c24aeb991e4b693f8fa37fc967079d3ec22 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/498#0183ce1d-c961-4259-8104-ed388d9679d5)
- 6bac4ab766d3c738bb816cb5fd6801aaab4388f4 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/495#0183cde1-fbb7-4898-be72-5dd19746ba96)
- 7d3eeb13032f341627015353214524fe09fad6a6 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/473#0183c981-3a1a-4474-9858-531a21b61568)
- 4b79a6f1d270658eca3bf65df2b16b4e512f6513 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/469#0183c942-db9c-448e-ba7f-94aa267aa8b3)
- c9875edafc167d70c55fef54cc045f14f39b19a5 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/468#0183c93d-5848-48df-827f-28864a880f55)
- 60840b5498a8e5e8fc6b1bbeadaaa5c210b62624 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/465#0183c8f6-c599-4cf6-9568-f8c7189d38db)
- 759b2d9970420e1e0c82c0ef83905f20a5c9fb28 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/462#0183c8ba-03e7-4348-91fe-849dd5fa2b0b)
- 297341e107daee1ea3aff991ae8ea8c90993c683 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/460#0183c862-44f4-46d2-a239-5a28b612f289)
- d89a66401719996b3f56b2769ca24d5fc20aa889 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/455#0183c827-acda-4d44-a279-c8796b077cb5)
- 217dcd7f18cd4ce0025a68bd46f0bf249c09870d FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/451#0183c4d7-a835-4349-b942-2f089c8f6800)
- 217dcd7f18cd4ce0025a68bd46f0bf249c09870d FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/454#0183c7fc-78db-441a-b5ab-50e0deea902b)
- 1bd8ab9b421a58cd07fe4ca015c07e5fd0070059 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/447#0183c270-61fd-40f4-8a3a-205e9ee71a66)
- 1bd8ab9b421a58cd07fe4ca015c07e5fd0070059 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/450#0183c4a6-b142-4a84-ac70-a346ce284d61)
- 9a7aa243aa6a74e146a7ce03d3e4d51a6917d43c FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/446#0183bd67-c7d4-495a-b34e-c5be679d67ac)
- ed06ceb68d42a02655b136d9cf8149f3a07b1231 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/444#0183b8e3-a8ff-4512-9592-11502d815941)
- d99eff919bf785f911e4eebc87ddc4960344a139 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/435#0183b57e-5ec9-46f4-9bfa-75f28342ad3d)
- 75a0c491e8ee4009e8828081059cfc98543c4d51 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/434#0183b534-cdb8-46ae-93cc-f936c81ac2dd)
- 3e1a5af990ead5554150578ace99bf05c207e084 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/433#0183b525-5514-43fa-a59c-82c65782347f)
- d9729f5fc9ae82bdb4471a609af4ab2d91fb3a38 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/431#0183b4ef-2604-4a9f-abe0-e23a1f044b3c)
- 5282ab49491fb8bf85b5b41405010245899368aa FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/430#0183b4b0-28aa-4c4d-a9a4-3642713f6fba)
- 815206ecc4c8cd8c288d0623e2cf27ac915fa5e7 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/429#0183b4a2-5433-450c-a344-59996bae4949)
- 47aa6c432353fb22a81103c28dc343b6d9407d52 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/428#0183b49d-8976-44a3-b635-2310fabdd33f)
- e16f49b327bbc1c18e8fc5d0ac4fa8c2f1144412 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/427#0183b48d-55a3-4c24-af63-54ca562e48b6)
- da6adbb5d6c692cc65d08a7cd4a72b4da72c4f93 FAILED [Buildkite :python: Minimal install 3.10](https://buildkite.com/ray-project/oss-ci-build-branch/builds/426#0183b48d-160f-4337-b4df-1eabd9be9310)
- 91040d91bba3cdfc29475b77e8d199884469ab86 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/379#0183aef3-a117-4bd9-afcb-2b2b95d4ec0b)
- 8486ddd1bf909449a22262c52a90f8359797fc16 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/378#0183aef2-533f-4b89-94ac-c90956576566)
- 7e939d79bcc68603b7c9a99fa6ef880dd526e931 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/377#0183aee5-bc1e-4fbf-ba49-994e11553f3d)
- 4732753327109d5c33b4cf69d375730057efc631 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/376#0183aeda-91f1-4c84-acca-e414fc5a1df4)
- 68a35fceef6a47fa3e3da41041ee32dafe7f24a8 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/375#0183aec2-9ad8-4058-9ae0-636b4dff5e3d)
- 23b3a599b9df8100558c477e94b0b19b1a38ac27 FAILED [Buildkite :python: Default install](https://buildkite.com/ray-project/oss-ci-build-branch/builds/374#0183aeb6-e8c8-4ed9-9710-a3af1ef8247d)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/dashboard:test_dashboard-END
....

### PR DESCRIPTION
**Pull request by rkooo567**
Signed-off-by: SangBin Cho <rkooo567@gmail.com>

## Why are these changes needed?

Trying out...

## Related issue number

Closes https://github.com/ray-project/ray/issues/29298

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Looks like the test still failing.

**Comment by rkooo567** [comment]
Now it passes. However, I think skipping test on Python 3.10 is not too bad idea too. (I think the impact of not enforcing version is pretty low)

File: dashboard/tests/test_dashboard.py [hunk 1]
```diff
@@ -1016,6 +1016,9 @@ def test_dashboard_module_load(tmpdir):
 def test_dashboard_module_no_warnings(enable_test_module):
     # Disable log_once so we will get all warnings
     from ray.util import debug
+    import six
```
Comment (rickyyx) [inline dashboard/tests/test_dashboard.py:1019]: is tis intentional left there?

**Review by rickyyx — CHANGES_REQUESTED** [review]
Left a few questions.

File: dashboard/modules/log/log_manager.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 logger = logging.getLogger(__name__)
 
-WORKER_LOG_PATTERN = re.compile(".*worker-([0-9a-f]+)-([0-9a-f]+)-(\d+).out")
+WORKER_LOG_PATTERN = re.compile(".*worker-([0-9a-f]+)-([0-9a-f]+)-(\\d+).out")
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/modules/log/log_manager.py:17]: I think it's broken without this.. but let me try

File: dashboard/tests/test_dashboard.py (hunk 1 above)
Comment (rkooo567) [reply dashboard/tests/test_dashboard.py:1019]: no lol. Thanks for catching it. No harto leave it though. But I will remove it

File: docker/ray-deps/Dockerfile [hunk 3]
```diff
@@ -13,7 +13,7 @@ RUN $HOME/anaconda3/bin/pip --no-cache-dir install --find-links $FIND_LINKS_PATH
     $(basename $WHEEL_PATH)[all] \
     $(if [ "$AUTOSCALER" = "autoscaler" ]; then echo \
         "redis>=3.5.0,<4.0.0" \
-        "six==1.13.0" \
+        "six==1.16.0" \
```
Comment (rkooo567) [inline docker/ray-deps/Dockerfile:16]: I might revert this

**Comment by rkooo567** [comment]
I think it is overkill this way. Alternative fix https://github.com/ray-project/ray/pull/29498

**Pull request closed without merging**